internal/users/manager_test.go

```diff
@@ -341,158 +324,6 @@ func TestRegisterUserPreauth(t *testing.T) {
 	}
 }
 
-func TestConcurrentUserUpdate(t *testing.T) {
```

Comment:
Well... This is the root of trust test to me. We need to pass this test, with or without global locking to be safe IMHO.

It can be improved, but we should not drop this, but rather use it as a way to check if things are working or not even in extreme situations.

They are likely nothing real, but being extreme is the only way we have to check that we are doing things properly even in the remote cases we did not consider.